codebase/src/java/disco/org/openlvc/disco/application/DeadReckoningModelStore.java

```diff
@@ -0,0 +1,208 @@
+/*
+ *   Copyright 2025 Open LVC Project.
+ *
+ *   This file is part of Open LVC Disco.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package org.openlvc.disco.application;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.openlvc.disco.application.deadreckoning.DeadReckoningModel;
+import org.openlvc.disco.pdu.entity.EntityStatePdu;
+import org.openlvc.disco.pdu.record.EntityId;
+
+/**
+ * Tracks objects and provides dead-reckoning models based on their last known state.
+ * As updates are received the previous model is replaced.
+ */
+public class DeadReckoningModelStore implements IDeleteReaperManaged
```

Comment:
Wondering whether this is similar enough to be merged with the EntityStore? If so you could just provide the following methods:

```
boolean isDeadReckoningAvailable( EntityId id );
DrmState getDeadReckonedStateAt( EntityId id, long localTimestamp );
DrmState getDeadReckonedStateAfter( EntityId id, double dt );
```

That would mean that all the entity related things are in one place, which makes things a bit simpler. Also dead reckoning calculations are only performed if you call the `getDeadReckonedStateXXX` methods, so its no additional overhead if the user don't want to utilize it.

Reply:
Had a chat to Nathan about this oob, and I we'll leave it like this for now. Argument for the current approach is that you can potentially DR more things than just entities, so planning ahead for the future it makes sense to have it on its own.

Reply:
Marking as resolved for now